core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -534,7 +534,8 @@ protected ObjectListingChunk getObjectListingChunkAndCreateNonEmpty(String path,
     String dir = stripPrefixIfPresent(path);
     ObjectListingChunk objs = getObjectListingChunk(dir, recursive);
     // If there are, this is a folder and we can create the necessary metadata
-    if (objs != null && objs.getObjectNames() != null && objs.getObjectNames().length > 0) {
+    if (objs != null && ((objs.getObjectNames() != null && objs.getObjectNames().length > 0)
+        || (objs.getCommonPrefixes() != null && objs.getCommonPrefixes().length > 0))) {
```

Comment:
There was a bug in which a directory with only subdirectories and no objects will not be discovered for s3a and likes. swift would have worked as it does not use commonPrefixes and combines prefixes in objectNames.